### PR TITLE
Core/Spells: Fixed possible use after free with deleted focusObject

### PR DESCRIPTION
**Pull request by Jildor**
**Changes proposed:**

-  Fixed possible use after free with deleted focusObject

**Issues addressed:**

None

**Tests performed:**

tested in-game

### CODE REVIEW

**Comment by jackpoz** [comment]
Is this not handled in UpdatePointer() or whatever is called at every update loop ?

**Comment by Jildor** [comment]
Updated and tested with feedback, thanks

File: src/server/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -7284,6 +7287,13 @@ bool Spell::UpdatePointers()
             m_originalCaster = nullptr;
     }
 
+    if (m_focusObjectGUID)
+    {
+        focusObject = ObjectAccessor::GetGameObject(*m_caster, m_focusObjectGUID);
+        if (focusObject && !focusObject->IsInWorld())
```
Comment (Shauren) [inline src/server/game/Spells/Spell.cpp:7293]: IsInWorld check is unneccessary, ObjectAccessor will never hand out pointers to objects not-in-world (other than players with FindConnectedPlayer)

Comment (Jildor) [reply src/server/game/Spells/Spell.cpp]: thanks, updated

**Comment by jackpoz** [comment]
Thanks for the PR 🚀 😃

**Pull request merged**